src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/prefermethodreference/InputPreferMethodReferenceMethodCallsOnExpression.java

```diff
@@ -0,0 +1,80 @@
+package com.puppycrawl.tools.checkstyle.checks.coding.prefermethodreference;
+
+import static java.lang.Long.sum;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.function.BiConsumer;
+import java.util.function.BiFunction;
+import java.util.function.Consumer;
+import java.util.function.Function;
+import java.util.function.Supplier;
+import javax.annotation.Nonnull;
+
+/**
+ * Config: detectForExpression = true
+ * Method calls cases
+ */
+public class InputPreferMethodReferenceMethodCallsOnExpression {
+    String field;
+    List<String> list;
+    // method calls
+    Consumer<Object> c = a -> System.out.println(a); // violation
+    BiConsumer<Long, Long> bc = (a, b) -> sum(a,b); // violation
+    BiConsumer<Long, Long> bc2 = (a, b) -> Long.sum(a,b); // violation
+    Function<Long, Long> f = a -> a.longValue(); // violation
+    Function<Long, Boolean> f4 = a -> super.equals(a); // violation
+    Runnable r = () -> System.out.println(); // violation
+    Function<Object, Object> f5 = a -> this.methodOneArg(a); // violation
+    BiFunction<String, String, Integer> f8 = (a,b) -> a.lastIndexOf(b); // violation
+    Function<String, String> f9 = concat -> field.concat(field).concat(concat); // violation
+    Function<String, String> f10 = a -> ((a)).trim(); // violation
+    Function<Integer, Integer> f11 = a -> new ArrayList<Integer>(){}.get(a); // violation
+    Supplier<Integer> f12 = () -> int[].class.hashCode(); // violation
+    Function<String, String> f13 = bool -> (bool() ? null : field).concat(bool); // violation
+    Function<String, String> f14 = a -> "das".concat(a); // violation
+    Function<String, Boolean> f15 =
+        (equals) -> (bool() ? list.removeIf(field::equals) : list).equals(equals); // violation
+    Function<String, String> f16 = field -> this.field.concat(field); // violation
+    Function<String, String> f17 = (String String) -> String.trim(); // violation
+    BiFunction<String, Long, Boolean> f19 =
+        (String String, Long Long) -> String.equals(Long); // violation
+    Function<Object, Integer> func = (@Nonnull Object obj) -> obj.hashCode(); // violation
+
+    // not violations
+    BiConsumer<Long, Long> callOnArg = (a, b) -> a.sum(a,b);
```

Comment:
let merge violation and no-valiation cases to more clearly see how code become better.

```
BiConsumer<Long, Long> ab = (a, b) -> sum(a,b); // violation
BiConsumer<Long, Long> callOnArg = (a, b) -> a.sum(a,b); // ok
```

There is not guaranty that static method has analog in instance method. Lets make `sum1` method and see behavior.

Reply:
Dont understand what you mean

Reply:
1. please merge `// not violations` and `// method calls` in same code block

2. Lets make Input class internal static method `sum1(a,b)`.
will be there violation `BiConsumer<Long, Long> ab = (a, b) -> sum1(a,b); // violation` ?